src/components/Nav/Nav.js

```diff
@@ -0,0 +1,151 @@
+    background: theme.palette.primary.main,
+  },
+});
+
+const tabsStyles = makeStyles({
+  root: {
+    width: '100%',
+    color: '#F2F3F2',
+  },
+});
+
+const iconStyles = makeStyles({
+  root: {
+    color: 'rgba(242, 243, 242, .5)',
+  },
+});
+
+const tabStyles = makeStyles({
+  root: {
+    borderRadius: '5px',
+    margin: '10px 15px',
+    '&:hover': {
+      backgroundColor: '#A6CF93',
+      cursor: 'pointer',
+      '& svg': {
+        color: 'white',
+      },
+    },
+    '&:focus': {
+      backgroundColor: '#A6CF93',
+      '& svg': {
+        color: 'white',
+      },
+    },
+  },
+});
+
+const IconTabs = (props) => {
+  const { match, history } = props;
+  const { params } = match;
+  const { page } = params;
+  const paperClasses = paperStyles();
+  const tabsClasses = tabsStyles();
+  const iconClasses = iconStyles();
+  const tabClasses = tabStyles();
+
+  const tabNameToIndex = {
+    0: props.token ? 'list' : 'welcome',
+    1: 'additems',
+    2: 'shareyourtoken',
+  };
+
+  const indexToTabName = {
+    list: 0,
+    welcome: 0,
+    additems: 1,
+    shareyourtoken: 2,
+  };
+
+  const [selectedTab, setSelectedTab] = useState(indexToTabName[page]);
+
+  const handleChange = (event, newValue) => {
+    history.push(`/${tabNameToIndex[newValue]}`);
+    setSelectedTab(newValue);
+  };
+
+  useEffect(() => {
+    setSelectedTab(indexToTabName[page]);
+  }, [page]);
+
+  return (
+    <>
+      <StyledDiv>
+        <Paper square className={paperClasses.root}>
+          <Tabs
+            className={tabsClasses.root}
+            value={selectedTab}
+            onChange={handleChange}
+            variant="fullWidth"
+            indicatorColor="primary"
+            textColor="primary"
+            aria-label="icon tabs example"
+            centered={true}
+          >
+            <Tab
+              className={tabClasses.root}
+              icon={<PlaylistAddCheckIcon className={iconClasses.root} />}
+              aria-label="list"
+            />
+            <Tab
+              className={tabClasses.root}
+              icon={<AddCircleOutlineIcon className={iconClasses.root} />}
+              aria-label="add item"
+            />
+            <Tab
+              className={tabClasses.root}
+              icon={<PeopleOutlineIcon className={iconClasses.root} />}
+              aria-label="share"
+            />
+          </Tabs>
+        </Paper>
+      </StyledDiv>
+      {selectedTab === 0 && props.token && (
+        <ItemList
+          list={props.list}
+          loading={props.loading}
+          error={props.error}
+          userToken={props.token}
+          history={history}
+        />
+      )}
+      {selectedTab === 0 && !props.token && (
+        <Welcome updateToken={props.updateToken} />
+      )}
+      {selectedTab === 1 && (
+        <AddItems list={props.list} userToken={props.token} />
+      )}
+      {selectedTab === 2 && <ShareYourToken history={history} />}
```

Comment:
Can you move this entire section above the `StyledDiv`? It messes with the tab order of the page having the content below the nav.  If you inspect the DOM, you'll see it actually shows the Nav tabs above the page content, and it's confusing when tabbing.

Reply:
Hey @laurenmbeatty! Great point. I also noticed this. Issue is that I'm not sure how to do this without having a function that retreives the `selectedTab` value back to Layout and render the components based on that. Normally that should be super easy to do but I can't figure out a way to pass the handler function from `Layout` down to the `{children}`. Any tips?